install/helm/superset/templates/deployment.yaml

```diff
@@ -42,6 +42,17 @@ spec:
           volumeMounts:
             - name: superset-config
               mountPath: /etc/superset/
+    {{- if .Values.supersetNode.hooks.postStart }}
+            - name: superset-config
+              mountPath: /superset_init.sh
+              subPath: superset_init.sh
+    {{- end }}
+          lifecycle:
+    {{- if .Values.supersetNode.hooks.postStart }}
+            postStart:
```

Comment:
Rather than using k8s hooks on the deployment level, you should instead spin a Job up which runs the init script. That way, users can safely tune their replica count up/down without re-triggering the init.

Reply:
Thanks, Jobs in Kubernetes will run to completion once they are created. I' will make configurable enable/disable of init job and in the init containers of job image I'll wait for db Do you see any drawback ?

Reply:
If a job fails another instance will be created until the job is completed or max retries reached. I think if the job fails because the database could not be reached and a new instance created after, this could be an alternative to adding logic in the container for waiting on db to be ready.